src/main/java/dev/openfeature/sdk/FeatureProviderWrapper.java

```diff
@@ -0,0 +1,118 @@
+package dev.openfeature.sdk;
+
+import dev.openfeature.sdk.exceptions.OpenFeatureError;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class FeatureProviderWrapper implements FeatureProvider, EventProviderListener {
+    private final FeatureProvider delegate;
+    private final AtomicBoolean isInitialized = new AtomicBoolean();
+    private ProviderState state = ProviderState.NOT_READY;
+
+    public FeatureProviderWrapper(FeatureProvider delegate) {
+        this.delegate = delegate;
+        if (delegate instanceof EventProvider) {
+            ((EventProvider) delegate).setEventProviderListener(this);
+        }
+    }
```

Comment:
Wrapping and delegating is a great solution here! Nice idea. It may be easier to do this with lombok's delegate though (see where we do it [here](https://github.com/open-feature/java-sdk/blob/c68f78e17b6125d74644a5735f9d58348edcc383/src/main/java/dev/openfeature/sdk/MutableContext.java#L24))